### PR TITLE
Calendar view

### PR DESCRIPTION
**Pull request by Prkhar05**
Added Calendar view according to #22

To see the Calendar View navigate to http://localhost:3000/events

### CODE REVIEW

File: src/components/Calendar/CalenderComponent/Month.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import Day from './Day'
+
+
+export default function Month({month}:{month:any}) {
+  return (
+    <div className='flex-1 grid grid-cols-7 grid-rows-5'>
+        {month.map((row: any[],i: any) => (
+            <React.Fragment>
```
Comment (web-team-iiti) [inline src/components/Calendar/CalenderComponent/Month.tsx:9]:
```suggestion
            <React.Fragment key={i}>
```

File: src/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React,{ useState , useContext ,useEffect } from "react";
+import { getMonth } from "./CalenderComponent/util";
+import Month from "./CalenderComponent/Month";
+import GlobalContext from "./context/GlobalContext";
+import CalendarHeader from "./CalenderComponent/CalendarHeader";
+import calenderSidebar from "./CalenderComponent/calenderSidebar";
```
Comment (web-team-iiti) [inline src/components/Calendar/Calendar.tsx:6]: where is this component? and is it being used in here?

File: src/components/Calendar/CalenderComponent/Month.tsx (hunk 1 above)
Comment (Prkhar05) [reply src/components/Calendar/CalenderComponent/Month.tsx]: Thanks for pointing out ,I've fixed it.

File: src/components/Calendar/Calendar.tsx (hunk 2 above)
Comment (Prkhar05) [reply src/components/Calendar/Calendar.tsx:6]: The Calendar.tsx is the final Calendar component which has been imported to \src\app\events\page.tsx for the purpose of displaying it on the web app.

Comment (web-team-iiti) [reply src/components/Calendar/Calendar.tsx:6]: talking about `import calenderSidebar from "./CalenderComponent/calenderSidebar";`

Comment (Prkhar05) [reply src/components/Calendar/Calendar.tsx:6]: Ok got it, yes its an unnecessary component which was used before , I will fix it.

File: src/components/Calendar/CalenderComponent/Day.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+export default function Day({day,rowIdx}:{day:any,rowIdx:any}) {
+  const [dayEvents,setDayEvents] = useState([])
+  const{ setShowEventModal , setDaySelected , filteredEvents , setSelectedEvent ,monthIndex }=
+  useContext(GlobalContext)
+
+  useEffect(() => {
+    const events = filteredEvents.filter((evt) => dayjs(evt.day).format("DD-MM-YY")  === day.format("DD-MM-YY"));
```
Comment (web-team-iiti) [inline src/components/Calendar/CalenderComponent/Day.tsx:16]: define type of `evt` here. it is reading type `never`

Comment (web-team-iiti) [reply src/components/Calendar/CalenderComponent/Day.tsx:16]: attaching error logs for reference

```
Type error: Property 'day' does not exist on type 'never'.
  14 |
  15 |   useEffect(() => {
> 16 |     const events = filteredEvents.filter((evt) => dayjs(evt.day).format("DD-MM-YY")  === day.format("DD-MM-YY"));
     |                                                             ^
  17 |     setDayEvents(events)
  18 |   },[filteredEvents,day]);
  19 |
```

Comment (Prkhar05) [reply src/components/Calendar/CalenderComponent/Day.tsx]: I have fixed it along with some other Type errors .

**Comment by harshtalati2410** [comment]
There seems to be no errors in current calendar.
Further changes and requirements:
1. The first row in any month is disabled. I am unable to click on it and create an event. Root cause being that the text in each day button is acting separately and hindering the button click.
2. Add more formats to the calendar. Include daily and weekly calendars if possible.

**Comment by Prkhar05** [comment]
> There seems to be no errors in current calendar. Further changes and requirements:
> 
> 1. The first row in any month is disabled. I am unable to click on it and create an event. Root cause being that the text in each day button is acting separately and hindering the button click.
> 2. Add more formats to the calendar. Include daily and weekly calendars if possible.

Sure , I will start working on that.

**Comment by Prkhar05** [comment]
I had unknowingly clicked on the "Update Branch" button. Did it cause any issue, If yes I apologize for that .How can undo any changes it might have caused?

**Comment by gjain-7** [comment]
> I had unknowingly clicked on the "Update Branch" button. Did it cause any issue, I apologize for that .How can I undo any changes it might have caused?

I think all you have done is synced your fork branch with the main branch, which has just merged main branch into your fork branch. This anyway would have been necessary to merge the pr. Just check if the current version of your fork branch works as intended.

**Comment by gjain-7** [comment]
@Prkhar05 What is the status of this PR? have you done the [requested changes](#issuecomment-2027943812)?

**Comment by Prkhar05** [comment]
> @Prkhar05 What is the status of this PR? have you done the [requested changes](#issuecomment-2027943812)?

Actually I got busy with my academics , but  the first change has been done , I am currently working on the daily and weekly calendar view for which I will require some more time to finish it.

**Comment by gjain-7** [comment]
> I am currently working on the daily and weekly calendar view for which I will require some more time to finish it.

no worries

**Review by mittal-ishaan — APPROVED** [review]

**Pull request merged**